x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditUtil.java

```diff
@@ -27,6 +27,9 @@ public class AuditUtil {
 
     public static String restRequestContent(RestRequest request) {
         if (request.hasContent()) {
+            if (request.isStreamedContent()) {
+                return "Request body had not been received at the time of the audit event";
```

Comment:
Do we want to eventually support this use case? If not, can we provide more details here? Something like
```
Audit logging with request body is not supported when the request is streamed. To disable request streaming, set [rest.incremental_bulk] to [false].
```
What do you think?

Reply:
Yes I think eventually we'll need this for streaming requests (because eventually we will be handling all requests as streaming). It's not reasonable to log an arbitrarily-large body in a single audit event tho, instead we will need to record each chunk in the audit log as they arrive.

Reply:
Also note that today because of how we try and log the whole body in a single message we end up truncating it anyway after a few kiB even with `rest.incremental_bulk: true`.

Reply:
Thanks for explaining. The fix here is certainly a lot better than throwing ClassCastException. But we may still want to create an issue to say a future fix is pending?

> we end up truncating it anyway after a few kiB 

IIRC, we don't truncate audit logs. At least payloads of a few hundred KB are fully logged.

Reply:
> IIRC, we don't truncate audit logs

You're right, TIL. And yet we seem to truncate other logs messages emitted by Log4J. I wonder why (but not hard enough to go digging further).